vpr/src/route/connection_router.cpp

```diff
@@ -702,7 +702,10 @@ void ConnectionRouter<Heap>::evaluate_timing_driven_node_costs(RTExploredNode* t
     bool switch_buffered = rr_switch_inf_[iswitch].buffered();
     bool reached_configurably = rr_switch_inf_[iswitch].configurable();
     float switch_R = rr_switch_inf_[iswitch].R;
-    float switch_Tdel = rr_switch_inf_[iswitch].Tdel;
+    // Instead of looking up the delay from the RR switch type info,
+    // we call edge_delay() method that returns the exact delay calculated
+    // by adding edge-specific delays to the nominal delay of switch type
```

Comment:
Test what the speed impact is of wrapping all the delay calculation functions in accessors, vs. getting an iswitch.
Might also be an opportunity to make the delay calculators more abstract.
Could make a method that returns a tuple of frequently used values in a hot loop if needed.

Reply:
Also think about whether we should take the Elmore delay model out of the router, and just have a linear delay model that is abstracted (ask for an edge delay, get a delay and you don't know how)